src/runtime/runtime.cs

```diff
@@ -1945,5 +1984,39 @@ internal static void SetNoSiteFlag()
                 }
             }
         }
+
+        /// <summary>
+        /// Return value: New reference.
+        /// </summary>
+        internal static IntPtr GetBuiltins()
+        {
+            return IsPython3 ? PyImport_ImportModule("builtins")
+                    : PyImport_ImportModule("__builtin__");
+        }
+    }
+
+
+    class PyReferenceCollection
+    {
+        private List<KeyValuePair<IntPtr, Action>> _actions = new List<KeyValuePair<IntPtr, Action>>();
+
+        /// <summary>
+        /// Record obj's address to release the obj in the future,
+        /// obj must alive before calling Release.
+        /// </summary>
+        public void Add(IntPtr ob, Action onRelease)
+        {
+            _actions.Add(new KeyValuePair<IntPtr, Action>(ob, onRelease));
```

Comment:
NIT: use `Tuple<T1, T2>`, e.g. just `_actions.Add((ob, onRelease))`

Reply:
Why using Tuple? As my experience, with two elements cases, KeyValue would always faster than tuple(unless the struct is too big).

Reply:
Simply for readability. This code is not a hot path, so a little performance difference hardly matters.
Also, once we retarget to newer framework, C# will use `ValueTuple` anyway.